src/main/kotlin/no/nav/familie/ef/iverksett/økonomi/simulering/kontroll/SimuleringskontrollService.kt

```diff
@@ -0,0 +1,134 @@
+package no.nav.familie.ef.iverksett.økonomi.simulering.kontroll
+
+import no.nav.familie.ef.iverksett.featuretoggle.FeatureToggleService
+import no.nav.familie.ef.iverksett.iverksetting.domene.IverksettData
+import no.nav.familie.ef.iverksett.iverksetting.domene.Simulering
+import no.nav.familie.ef.iverksett.iverksetting.domene.tilSimulering
+import no.nav.familie.ef.iverksett.økonomi.simulering.SimuleringService
+import no.nav.familie.kontrakter.felles.simulering.BeriketSimuleringsresultat
+import no.nav.familie.kontrakter.felles.simulering.Simuleringsperiode
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+import java.time.YearMonth
+import java.util.UUID
+
+@Service
+class SimuleringskontrollService(
+    private val simuleringService: SimuleringService,
+    private val simuleringskontrollRepository: SimuleringskontrollRepository,
+    private val featureToggleService: FeatureToggleService,
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    fun kontrollerMedNyUtbetalingsgenerator(
+        iverksett: IverksettData,
+        beriketSimuleringsresultatFn: () -> BeriketSimuleringsresultat,
+    ) {
+        if (!featureToggleService.isEnabled("familie.ef.iverksett.simuleringskontroll")) {
+            return
+        }
+        if (iverksett.behandling.forrigeBehandlingId == null) {
+            return
```

Comment:
Hvis noen har migrert fra infotrygd så vil vel "forrigeBehandlingId" være null, men fremdeles aktuelt med simulering, eller?

Reply:
Tenkte at kontrollen var mest interessant for revurderinger
Hvis man har migrert fra infotrygd, og man iverksetter en ny behandling så har man forrigeBehandlingId som er migreringen

Reply:
Ja, sant. og vi kommer jo ikke til å migrere all verden fremover  :)